libmachine/provision/alpine.go

```diff
@@ -0,0 +1,173 @@
+package provision
+
+import (
+	"fmt"
+	"github.com/docker/machine/libmachine/auth"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/engine"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/docker/machine/libmachine/provision/pkgaction"
+	"github.com/docker/machine/libmachine/provision/serviceaction"
+	"github.com/docker/machine/libmachine/swarm"
+)
+
+func init() {
+	Register("Alpine", &RegisteredProvisioner{
+		New: NewAlpineProvisioner,
+	})
+}
+
+func NewAlpineProvisioner(d drivers.Driver) Provisioner {
+	return &AlpineProvisioner{
+		GenericProvisioner{
+			SSHCommander:      GenericSSHCommander{Driver: d},
+			DockerOptionsDir:  "/etc/docker",
+			DaemonOptionsFile: "/etc/conf.d/docker",
+			OsReleaseID:       "alpine",
+			Packages: []string{
+				"curl",
+			},
+			Driver: d,
+		},
+	}
+}
+
+type AlpineProvisioner struct {
+	GenericProvisioner
+}
+
+func (provisioner *AlpineProvisioner) String() string {
+	return "alpine"
+}
+
+func (provisioner *AlpineProvisioner) CompatibleWithHost() bool {
+	return provisioner.OsReleaseInfo.ID == provisioner.OsReleaseID || provisioner.OsReleaseInfo.IDLike == provisioner.OsReleaseID
+}
+
+func (provisioner *AlpineProvisioner) Package(name string, action pkgaction.PackageAction) error {
+	var packageAction string
+
+	switch action {
+	case pkgaction.Install, pkgaction.Upgrade:
+		packageAction = "add"
+	case pkgaction.Remove:
+		packageAction = "remove"
```

Comment:
```
$ apk remove
ERROR: 'remove' is not an apk command. See 'apk --help'.
```

This should be:
`packageAction = "del"`

Reply:
Yes, thanks